test/test_datasets.py

```diff
@@ -944,14 +943,44 @@ def test_not_found_or_corrupted(self):
             super().test_not_found_or_corrupted()
 
 
+class KineticsTestCase(datasets_utils.VideoDatasetTestCase):
+    DATASET_CLASS = datasets.Kinetics
+    ADDITIONAL_CONFIGS = datasets_utils.combinations_grid(
+        split=("train", "val"), num_classes=("400", "600", "700")
+    )
+
+    def inject_fake_data(self, tmpdir, config):
+        classes = ("Abseiling", "Zumba")
+        num_videos_per_class = 2
+        tmpdir = pathlib.Path(tmpdir) / config['split']
+        digits = string.ascii_letters + string.digits + "-_"
+        for cls in classes:
+            datasets_utils.create_video_folder(
+                tmpdir,
+                cls,
+                lambda _: f"{datasets_utils.create_random_string(11, digits)}.mp4",
+                num_videos_per_class,
+            )
+        return num_videos_per_class * len(classes)
+
+
 class Kinetics400TestCase(datasets_utils.VideoDatasetTestCase):
     DATASET_CLASS = datasets.Kinetics400
 
+    def dataset_args(self, tmpdir, config):
+        # note: train is here hardcoded by default bc we expect the user to supply it,
+        # but that requirement have changed in subsequent version of the dataset
+        root = pathlib.Path(tmpdir) / "train"
+        return root, 1
+
     def inject_fake_data(self, tmpdir, config):
         classes = ("Abseiling", "Zumba")
         num_videos_per_class = 2
 
         digits = string.ascii_letters + string.digits + "-_"
+        # note: train is here hardcoded by default bc we expect the user to supply it,
+        # but that requirement have changed in subsequent version of the dataset
+        tmpdir = pathlib.Path(tmpdir) / "train"
```

Comment:
We should aim for BC. That means we shouldn't need to change anything here and the tests should pass.